sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -301,6 +265,8 @@ class ProjectOverviewView extends VerticalLayout implements HasHotbar, HasTitle
     }
 
     private void fillProjectsGrid() {
+        projectGrid.addColumn({ viewModel.projectToCheckbox.get(it.code)}, new ComponentRenderer())
```

Comment:
Instead of reading from the map directly, it would make sense to use the factory here. The factory should do the lookup IMHO so that you can lazily fill the hashmap. 
That would be the advantage of the factory. It does not have to be known whether a checkbox for the project already exists or not.

Reply:
see my comment below. It makes no sense to me to make a factory, which produces the checkbox also aware of what it has already produced. Then its more like a container or a "view component" since it contains the view elements.

Reply:
https://refactoring.guru/design-patterns/factory-method <- It is a well known pattern that elements can be reused using the factory pattern.
> Use the Factory Method when you want to save system resources by reusing existing objects instead of rebuilding them each time